apis/extension/priority_qos.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extension
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type PriorityQoSClass string
+
+const (
+	PriorityQoSSeparator   = "/"
+	PriorityQoSClassSystem = PriorityQoSClass(QoSSystem)
+	PriorityQoSClassNone   = PriorityQoSClass("")
+)
+
+// Supported combinations:
+// SYSTEM
+// koord-prod: LS LSE LSR
+// koord-mid: LS BE
+// koord-batch: BE
```

Comment:
Batch + LS, Free + LS are acceptable in some scenarios.
e.g. Apps allocate Batch resources to co-locate with the Prod pods, but they'd rather get evicted than suppressed.

For now, it is suggested that check the combinations of the priority classes and QoS classes loosely.